### PR TITLE
WIP: Dockerfile

### PR DESCRIPTION
**Pull request by steinbro**
First task in #4. This currently builds, but doesn't run due to a missing auth file. Where do I get one of these? I don't understand the language int the README.

### CODE REVIEW

**Comment by sheldonej** [comment]
> First task in #4. This currently builds, but doesn't run due to a missing auth file. Where do I get one of these? I don't understand the language int the README.

Add a folder named ".auth" (the folder needs the period in the name) and a json file name "me.json" inside of that folder. The json can be empty: {}. This should fix that issue.


The structure of that json is followed if you want to fill it out with test data for the time being. 

[
    {
      "user_claims": [
        {
          "typ": "http://schemas.microsoft.com/identity/claims/objectidentifier",
          "val": "123vhjvjh45"
        },
        {
          "typ": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
          "val": "users@example.com" 
        },
        {
          "typ": "name",
          "val": "John Doe" 
        },
        {
          "typ": "preferred_username",
          "val": "johndoe" 
        }
    ],
      "id_token": "some_id_token_value"
    }
  ]

**Comment by RDMurray** [comment]
@steinbro what do you want to do with this pr? We should probably get it up to date so it can be used for production. I'm happy to take it on if you don't have time.

**Comment by steinbro** [comment]
I was assuming you were already using something of this sort for the current deployment, but if this is still useful then by all means update and merge it in. It's been long enough since I've touched it that you're probably better suited to pickit up than I.

**Comment by RDMurray** [comment]
Ok I think this might be ready for production. It seems to be working. I'll get it up on a temporary host name for testing.

@steinbro would you like to review before I merge?

File: backend/backend/settings/base.py [hunk 1]
```diff
@@ -155,7 +155,7 @@
 STATIC_ROOT = os.path.join(BASE_DIR, "staticfiles")
 
 # This section is added to support serving the frontend files
-#INSTALLED_APPS.extend(["whitenoise.runserver_nostatic"])
+INSTALLED_APPS.extend(["whitenoise.runserver_nostatic"])
```
Comment (steinbro) [inline backend/backend/settings/base.py:158]: This either belongs directly in the definition of `INSTALLED_APPS` above (line 48), or, if it's only needed for e.g. production, in the respective settings file for that environment.

File: backend/backend/settings/production.py [hunk 2]
```diff
@@ -25,3 +25,13 @@
...
+
+# Used for serving local user-uploaded files
+# I.g., "http://127.0.0.1:8000/files/{file_path}"
+MEDIA_URL = 'files/'
+
+# Used for storing local user-uploaded files
+MEDIA_ROOT = os.path.join(BASE_DIR, MEDIA_URL)
+
+
+ALLOWED_HOSTS = os.environ['ALLOWED_HOSTS'].split(' ')
```
Comment (steinbro) [inline backend/backend/settings/production.py:37]: Why is `ALLOWED_HOSTS` space-delimited in production, overriding the comma-delimited default in the base settings?

**Review by steinbro — COMMENTED** [review]
No objections from me. I didn't test it myself, but left a few minor comments.

File: backend/backend/settings/base.py (hunk 1 above)
Comment (RDMurray) [reply backend/backend/settings/base.py:158]: I agree. To be honest I don't know if it is needed at all. It was in the original MS code and I think I commented it out at some point. If it works I'll revert this change and we can figure out if whitenoise is even necessary in the future.

File: backend/backend/settings/production.py (hunk 2 above)
Comment (RDMurray) [reply backend/backend/settings/production.py:37]: Oops, no reason, I didn't even look to see if it was already defined in base.py.

**Pull request merged**